src/IssueViz/Editor/ErrorTagging/ErrorTagTooltipProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Infrastructure.VS;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.ErrorTagging
+{
+    /// <summary>
+    /// Creates a tooltip object for an IErrorTag
+    /// </summary>
+    internal interface IErrorTagTooltipProvider
+    {
+        object Create(IAnalysisIssueBase analysisIssueBase);
+    }
+
+    [Export(typeof(IErrorTagTooltipProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ErrorTagTooltipProvider : IErrorTagTooltipProvider
+    {
+        private readonly IBrowserService browserService;
+        private readonly IVsThemeColorProvider vsThemeColorProvider;
+        private readonly IRuleHelpLinkProvider ruleHelpLinkProvider;
+
+        [ImportingConstructor]
+        public ErrorTagTooltipProvider(IBrowserService browserService, IVsThemeColorProvider vsThemeColorProvider)
+            : this(browserService, vsThemeColorProvider, new RuleHelpLinkProvider())
+        {
+        }
+
+        internal ErrorTagTooltipProvider(IBrowserService browserService, IVsThemeColorProvider vsThemeColorProvider, IRuleHelpLinkProvider ruleHelpLinkProvider)
+        {
+            this.browserService = browserService;
+            this.vsThemeColorProvider = vsThemeColorProvider;
+            this.ruleHelpLinkProvider = ruleHelpLinkProvider;
+        }
+
+        public object Create(IAnalysisIssueBase analysisIssueBase)
+        {
+            var ruleUrl = ruleHelpLinkProvider.GetHelpLink(analysisIssueBase.RuleKey);
+
+            var hyperLink = new Hyperlink
```

Comment:
The end result looks ok, but for the life of me - I couldn't add a style that adds an underline when hovering over the link (which is the default VS behavior for the built in tooltips).
So the link is blue but there is no hover effect. I've spent an hour on this and in the ended I decided to drop it.
Let me know if you can think of a way.

Reply:
I think it's good enough to ship as it is. 

What did you try? I'd have thought using a style trigger bound to the `IsMouseOver` property of the hyperlink would have worked, perhaps with two versions of the `hyperlink.Inlines`, one with an `Underline` run and one without.

Reply:
That's what I tried, it doesn't work. I'll leave it as-is.